### PR TITLE
refactor(esx_society/client/main.lua) : Fixed stuff in boss menu and added translations

### PR DESCRIPTION
**Pull request by lexinor**
- Added a possibility to return from almost every part in the boss menu, that's why we are now passing "options" through the whole menu to be able to come back at the initial state.

- Added a null check when changing a job grade's name by "nil" and did the same for job ranks.

- Added some new translations. I only translated FR and EN, others are just plain english values

### CODE REVIEW

File: [esx_addons]/esx_society/locales/it.lua [hunk 1]
```diff
@@ -12,19 +12,17 @@ Locales['it'] = {
   ['grade']                  = 'grado',
   ['have_deposited']         = 'hai depositato ~r~$%s',
   ['have_withdrawn']         = 'hai prelevato $%s',
-  ['check_balance']          = 'il bilancio è ~g~$%s',
   ['invalid_amount']         = 'importo non valido',
-  ['invalid_amount_max']     = 'quel salario non è consentito',
+  ['invalid_amount_max']     = 'quel salario non è consentito',  
+  ['invalid_value']          = 'invalid value',
```
Comment (bitpredator) [inline [esx_addons]/esx_society/locales/it.lua:17]: ['invalid_value']  = 'valore non valido',

File: [esx_addons]/esx_society/locales/it.lua [hunk 2]
```diff
@@ -12,19 +12,17 @@ Locales['it'] = {
   ['grade']                  = 'grado',
   ['have_deposited']         = 'hai depositato ~r~$%s',
   ['have_withdrawn']         = 'hai prelevato $%s',
-  ['check_balance']          = 'il bilancio è ~g~$%s',
   ['invalid_amount']         = 'importo non valido',
-  ['invalid_amount_max']     = 'quel salario non è consentito',
+  ['invalid_amount_max']     = 'quel salario non è consentito',  
+  ['invalid_value']          = 'invalid value',  
+  ['invalid_value_nochanges']= 'invalid value, no changes applied',
```
Comment (bitpredator) [inline [esx_addons]/esx_society/locales/it.lua:18]: ['invalid_value_nochanges']= 'valore non valido, nessuna modifica applicata',

**Review by bitpredator — CHANGES_REQUESTED** [review]
I have corrected the new translations inserted regarding the Italian language

File: [esx_addons]/esx_society/locales/it.lua (hunk 1 above)
Comment (bitpredator) [inline [esx_addons]/esx_society/locales/it.lua:17]:
```suggestion
  ['invalid_value']          = 'valore non valido',  
```

File: [esx_addons]/esx_society/locales/it.lua (hunk 2 above)
Comment (bitpredator) [inline [esx_addons]/esx_society/locales/it.lua:18]:
```suggestion
  ['invalid_value_nochanges']= 'valore non valido, nessuna modifica applicata',
```

**Review by bitpredator — CHANGES_REQUESTED** [review]

File: [esx_addons]/esx_society/locales/it.lua (hunk 1 above)
Comment (lexinor) [reply [esx_addons]/esx_society/locales/it.lua:17]: Thanks for your participation !

**Review by lexinor — COMMENTED** [review]
Include that translation

**Comment by lexinor** [comment]
@Benzo00 Any news ?

**Comment by lexinor** [comment]
@JackDUpModZ @Benzo00 Could you review it please ? I need it merged

**Review by TrymTube — APPROVED** [review]

**Review by bitpredator — APPROVED** [review]

**Comment by Gellipapa** [comment]
@lexinor Thank you for pull request.  Sorry for the long wait, there will soon be changes in the acceptance of PRs. I tested it and there are no bugs in it but I rewrote two things on the server side. Soon to be merged.

**Review by Gellipapa — APPROVED** [review]
Tested, works.

**Comment by lexinor** [comment]
> @lexinor Thank you for pull request.  Sorry for the long wait, there will soon be changes in the acceptance of PRs. I tested it and there are no bugs in it but I rewrote two things on the server side. Soon to be merged. 

Thanks for your response, I hope it will be faster next time 😅😅.
There are still some work to do on the menu anyway so glad to see that

**Comment by Gellipapa** [comment]
Yes, for exampl outsource all burned-in menu text to a local file.

**Pull request merged**